include/clap/helpers/host-proxy.hh

```diff
@@ -8,8 +8,215 @@
 #include "misbehaviour-handler.hh"
 
 namespace clap { namespace helpers {
+   // clap_host_log
    template <MisbehaviourHandler h, CheckingLevel l>
-   class HostProxy {
+   class HostLogProxy {
+   public:
+      virtual bool canUseHostLog() const noexcept = 0;
```

Comment:
OK here's my primary concern. 

You've made these functions virtual even though we are in a template space and I'm not entirely sure why.  But I also think that's probably a mistake to add the virtual dispatch here.

If you just remove `virtual`, `=0` and `override` everywhere, then make the .hxx have the appropriate base class names, does this still work?

I'm also still having a hard time seeing exactly the usage pattern implied, but 'make everything virtual rather than concrete when we are already in an explicit template specialization space' seems like it requires some thought.

Maybe for me to fully understand this we need to write a short program which uses one of these internal proxies directly in a way which requires virtual dispatch as opposed to just base classing with a concrete type. Does that make sense?

Reply:
I tried to make the base classes pure interfaces (but in case of leaving it like this should add virtual destructors to them! see https://stackoverflow.com/questions/318064/how-do-you-declare-an-interface-in-c#318137)

To remove virtual, it would be a totally different concept. they wouldn't be interfaces anymore and I'd have to move the extension pointers into them and they'd need to hold a reference to the `clap_plugin`/`clap_host` each. And then they would not have to be base classes of the PluginProxy/HostProxy anymore but could be members replacing the extension pointer members.

Or do I see this wrong?

Reply:
> Maybe for me to fully understand this we need to write a short program which uses one of these internal proxies directly in a way which requires virtual dispatch as opposed to just base classing with a concrete type. Does that make sense?

So I probably should add tests for showcasing (and verifying) the usage

Reply:
Just thought about this whole topic again: It really might not be a good idea to enforce virtual here.

If someone really wants to pass only part of the PluginProxy/HostProxy methods, it can be done by creating a class that holds a reference to the PluginProxy/HostProxy and offers a reduced set of public methods. Or someone derives from a protected/private PluginProxy/HostProxy and offers public methods as desired